src/CowEvcOpenPositionWrapper.sol

```diff
@@ -0,0 +1,215 @@
+        address owner;
+
+        /// @dev The subaccount to open the position on. Learn more about Euler subaccounts https://evc.wtf/docs/concepts/internals/sub-accounts
+        address account;
+
+        /// @dev A date by which this operation must be completed
+        uint256 deadline;
+
+        /// @dev The Euler vault to use as collateral
+        address collateralVault;
+
+        /// @dev The Euler vault to use as leverage
+        address borrowVault;
+
+        /// @dev The amount of collateral to import as margin. Set this to `0` if the vault already has margin collateral.
+        uint256 collateralAmount;
+
+        /// @dev The amount of debt to take out. The borrowed tokens will be converted to `collateralVault` tokens and deposited into the account.
+        uint256 borrowAmount;
+    }
+
+    function _parseOpenPositionParams(bytes calldata wrapperData)
+        internal
+        pure
+        returns (OpenPositionParams memory params, bytes memory signature)
+    {
+        (params, signature) = abi.decode(wrapperData, (OpenPositionParams, bytes));
+    }
+
+    /// @notice Helper function to compute the hash that would be approved
+    /// @param params The OpenPositionParams to hash
+    /// @return The hash of the signed calldata for these params
+    function getApprovalHash(OpenPositionParams memory params) external view returns (bytes32) {
+        return _getApprovalHash(memoryLocation(params));
+    }
+
+    /// @inheritdoc CowWrapper
+    function validateWrapperData(bytes calldata wrapperData) external pure override {
+        // Validate by attempting to parse the wrapper data
+        // Will revert if the data is malformed
+        _parseOpenPositionParams(wrapperData);
+    }
+
+    /// @notice Implementation of CowWrapper._wrap - executes EVC operations to open a position
+    /// @param settleData Data which will be used for the parameters in a call to `CowSettlement.settle`
+    /// @param wrapperData Additional data containing OpenPositionParams
+    function _wrap(bytes calldata settleData, bytes calldata wrapperData, bytes calldata remainingWrapperData)
+        internal
+        override
+    {
+        // Decode wrapper data into OpenPositionParams
+        (OpenPositionParams memory params, bytes memory signature) = _parseOpenPositionParams(wrapperData);
+
+        _invokeEvc(
+            settleData,
+            wrapperData,
+            remainingWrapperData,
+            memoryLocation(params),
+            signature,
+            params.owner,
+            params.deadline
+        );
+
+        emit CowEvcPositionOpened(
+            params.owner,
+            params.account,
+            params.collateralVault,
+            params.borrowVault,
+            params.collateralAmount,
+            params.borrowAmount
+        );
+    }
+
+    /// @notice Called by an offchain process to determine what data should be signed in a call to `wrappedSettle`.
```

Comment:
The comment is too generic, this function has a very specific purpose. 

```suggestion
    /// @notice Called by an offchain process to determine what data should be signed by users in case they choose to use the permit flow for authentication.
```